src/services/CertificateGenerationService.ts

```diff
@@ -230,22 +225,22 @@ class CertificateGenerationService {
     const secretConfig = await this.getSecret();
 
     if (secretConfig) {
-      const addressResponse: boolean = await axios({
-        method: "get",
-        url: secretConfig.url + "/" + postcode,
-        headers: {"x-api-key": secretConfig.key},
-      })
-          .then((response) => {
-            return response.data.isWelshAddress;
-          })
-          .catch((error) => {
-            console.log(`Error looking up postcode ${postcode}`);
-            console.log(error);
-            return false;
-          });
-
-      console.log(`Return value for isWelsh for ${postcode} is ${addressResponse}`);
-      return addressResponse;
+        let retries = 0;
+        while (retries < 3) {
+            try {
+                const addressResponse = await axios({
+                    method: "get",
+                    url: secretConfig.url + "/" + postcode,
+                    headers: {"x-api-key": secretConfig.key},
+                });
+                return addressResponse.data.isWelshAddress;
+            } catch (error) {
+                retries++;
+                console.log(`Error looking up postcode ${postcode}`);
```

Comment:
Can you include the attempt number on the error log, will help with support and to identify if the retry is working as expected